### PR TITLE
Converted to Keras Core: A Vision Transformer without Attention

### PR DESCRIPTION
**Pull request by anas-rz**
Remove tensorflow_addons dependency
Fix imports
Update shape computation in `DropPath` to `shape = (tf.shape(x)[0],) + (1,) * (len(x.shape) - 1)`
Update loss computation metrics update in subclassed model
Linked Issue: #222

### CODE REVIEW

File: examples/keras_io/tensorflow/vision/shiftvit.py [hunk 1]
```diff
@@ -0,0 +1,880 @@
+
```
Comment (fchollet) [inline examples/keras_io/tensorflow/vision/shiftvit.py:1]: Remove blank line

File: examples/keras_io/tensorflow/vision/shiftvit.py [hunk 2]
```diff
@@ -0,0 +1,880 @@
...
+In the academic paper
+[When Shift Operation Meets Vision Transformer: An Extremely Simple Alternative to Attention Mechanism](https://arxiv.org/abs/2201.10801)
+the authors propose to demystify the success of ViTs with the introduction of a **NO
+PARAMETER** operation in place of the attention operation. They swap the attention
+operation with a shifting operation.
+
+In this example, we minimally implement the paper with close alignement to the author's
+[official implementation](https://github.com/microsoft/SPACH/blob/main/models/shiftvit.py).
+
+
```
Comment (fchollet) [inline examples/keras_io/tensorflow/vision/shiftvit.py:31]: You can remove empty text lines at the end of text blocks

**Comment by anas-rz** [comment]
Here it is.
[shiftvit_keras_core-diff.txt](https://github.com/keras-team/keras-core/files/12073770/shiftvit_keras_core-diff.txt)

**Comment by fchollet** [comment]
Thanks! The number of epochs should be reverted to its original value.

**Comment by anas-rz** [comment]
Done.

**Review by fchollet — APPROVED** [review]
LGTM, thank you!
